src/map/campaign_system.h

```diff
@@ -0,0 +1,183 @@
+/*
+===========================================================================
+
+Copyright (c) 2010-2015 Darkstar Dev Teams
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see http://www.gnu.org/licenses/
+
+This file is part of DarkStar-server source code.
+
+===========================================================================
+*/
+
+#ifndef _CAMPAIGNSYSTEM_H
+#define _CAMPAIGNSYSTEM_H
+
+#include <vector>
+#include "../common/cbasetypes.h"
+
+enum CampaignControl : uint8
+{
+    SandoriaMask = 0x02,
+    BastokMask = 0x04,
+    WindurstMask = 0x06,
+    BeastmanMask = 0x08
+};
+
+enum CampaignZone : uint8
```

Comment:
should also have some way of getting this value from the regular zoneid.  a column in zone_settings, maybe, and then a local on the CZone

Reply:
I thought about it. I was going to remove the enum completely, as the core only uses it for type information, but I haven't yet confirmed if it is used in any of the troop movement messages. Once I confirm that, the plan is to change ID to be the zoneid and add a new column for sort order.